verify_mapping_quality.pl

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+use Pod::Usage;
+use Getopt::Long;
+use autodie;
+use FindBin qw($Bin);
+use File::Basename;
+use Parallel::ForkManager;
+use Readonly;
+
+Readonly my $MIN_DEPTH => 10;    # THIRD COLUMN
+Readonly my $MIN_MAP => 80;
+__PACKAGE__->run unless caller;
+
+1;
+
+sub run {
+    my ( $dir, $size, $man, $help, $min_depth, $log_dir, $min_map );
+
+    GetOptions(
+        "i|dir=s"     => \$dir,
+        "l|log_dir=s" => \$log_dir,
+        "s|size=s"    => \$size,
+        "min-map=f"	  => \$min_map,
+        "min-depth=i" => \$min_depth,
+        "h|help"      => \$help,
+        "m|man"       => \$man
+    );
+    pod2usage(1) if $help;
+    pod2usage(-verbose => 2) if $man;
+
+    unless ( (defined $dir && -e $dir) ) {
+        print "Please specifiy a directory or file name.\n\n";
+        pod2usage(1);
+    }
+    unless ( defined $size ) {
+        print "Please specify the size of the genome.\n\n";
+        pod2usage(1);
+    }
+
+	#set default values if undefined on command line
+    if(!defined $min_depth){
+        $min_depth = $MIN_DEPTH;
+        print "  Using a minumum depth of $MIN_DEPTH\n";
+        print "  Use the --min-depth flag to set custom depth\n\n";
+    }
+    else{
+    	print "Using a min_depth of ".$min_depth."\n";
+    }
+	$log_dir='' if (not defined $log_dir);
+	$min_map=$MIN_MAP if (not defined $min_map);
+	
+	#create the log file to print warnings to
+	open(my $log, '>'.$log_dir.'mapping_percentage.log');
+	
+    my @files;
+    @files = get_bams($dir);
+	
+	#parse results and determine if the pipeline should die, or warnings thrown
+	my @results;
+    @results = verify_percent_coverage( \@files, $size, $min_depth );
+    
+    print $log "==========Reference Mapping Quality===========\n";
+    print $log "NUMBER OF BP's IN REFERENCE GENOME: ".$size."\n";
+    foreach my $result(@results){
+    	my @split = split(',', $result);
+    	my @double = split('%', $split[1]);
+    	print $log "Mapping to reference for isolate ".$split[0]." is ".$split[1]."\n" if $double[0] < $min_map; 
+    }
+   	
+}
+
+#----------------------------------------------------------#
+# Calculate the % Coverage for all .bam files in directory #
+#----------------------------------------------------------#
+sub verify_percent_coverage {
+    my ( $files, $size, $min_depth ) = @_;
+
+    # Get the number of available cores
+    #----------------------------------#
+    my $cores = get_num_cores();
```

Comment:
Get rid of this code and the number of core to be used should be given by the user on the command line. Otherwise, the script will take more then it was too.